### PR TITLE
GRU + other small things

### PR DESCRIPTION
**Pull request by ssss1029**
- Add sanity check to dataloading: Error if there is a file with a function start and end on the same byte. (So we can know of this if it ever occurs in the future; I didn't find any in the elf_x64 data)
 - Add GRU training script.
 - Add F1 score util function

### CODE REVIEW

**Comment by xksteven** [comment]
The train_rnn shouldn't really exist.  Any time there is significant overlap (as in lots of copy/paste) then the code should be become it's own function or module etc.

**Comment by xksteven** [comment]
I'll wait until you commit this before I work on the loss function/GRU to see where you leave it off at.  I'll keep working on data processing part.
